xbmc/video/windows/GUIWindowVideoBase.cpp

```diff
@@ -541,12 +541,14 @@ bool CGUIWindowVideoBase::OnSelect(int iItem)
   CFileItemPtr item = m_vecItems->Get(iItem);
 
   std::string path = item->GetPath();
-  if (!item->m_bIsFolder && path != "add" &&
-      !StringUtils::StartsWith(path, "newsmartplaylist://") &&
-      !StringUtils::StartsWith(path, "newplaylist://") &&
-      !StringUtils::StartsWith(path, "newtag://") && !StringUtils::StartsWith(path, "script://") &&
-      !StringUtils::StartsWith(path, "plugin://") && !item->HasProperty("IsPlayable") &&
-      !item->GetProperty("IsPlayable").asBoolean())
+  if ((!item->m_bIsFolder && path != "add" &&
+       (!StringUtils::StartsWith(path, "newsmartplaylist://") &&
+        !StringUtils::StartsWith(path, "newplaylist://") &&
+        !StringUtils::StartsWith(path, "newtag://") &&
+        !StringUtils::StartsWith(path, "script://") &&
+        !StringUtils::StartsWith(path, "plugin://"))) ||
+      (StringUtils::StartsWith(path, "plugin://") && item->HasProperty("IsPlayable") &&
+       item->GetProperty("IsPlayable").asBoolean()))
```

Comment:
Works but a bit hard to read for me, what about this? (group the plugin stuff and use the asBoolean default value)
```suggestion
  if ((!item->m_bIsFolder && path != "add" &&
       (!StringUtils::StartsWith(path, "newsmartplaylist://") &&
        !StringUtils::StartsWith(path, "newplaylist://") &&
        !StringUtils::StartsWith(path, "newtag://") &&
        !StringUtils::StartsWith(path, "script://") &&
        (!StringUtils::StartsWith(path, "plugin://") ||
         (StringUtils::StartsWith(path, "plugin://") &&
          item->GetProperty("IsPlayable").asBoolean(false))))))
```

Reply:
actually there is an extra top level of parenthesis that's not needed in my suggestion, should be
````suggestion
  if (!item->m_bIsFolder && path != "add" &&
      (!StringUtils::StartsWith(path, "newsmartplaylist://") &&
       !StringUtils::StartsWith(path, "newplaylist://") &&
       !StringUtils::StartsWith(path, "newtag://") && !StringUtils::StartsWith(path, "script://") &&
       (!StringUtils::StartsWith(path, "plugin://") ||
        (StringUtils::StartsWith(path, "plugin://") &&
         item->GetProperty("IsPlayable").asBoolean(false)))))
```